packages/services/src/services/VideoPlayer/VideoPlayer.js

```diff
@@ -182,18 +190,23 @@ class VideoPlayerAPI {
    */
   static async api(videoId) {
     return await this.checkScript().then(() => {
-      return new Promise(resolve => {
-        return new root.kWidget.api({ wid: _partnerId }).doRequest(
-          {
-            service: 'media',
-            action: 'get',
-            entryId: videoId,
-          },
-          function(jsonObj) {
-            resolve(jsonObj);
-          }
-        );
-      });
+      if (videoData && videoData.id === videoId) {
+        return videoData;
+      } else {
+        return new Promise(resolve => {
+          return new root.kWidget.api({ wid: _partnerId }).doRequest(
+            {
+              service: 'media',
+              action: 'get',
+              entryId: videoId,
+            },
+            function(jsonObj) {
+              videoData = jsonObj;
```

Comment:
should this be set actually as an associative array in case multiple videos are on the page?
```videoData[jsonObj.id] = jsonObj;```